### PR TITLE
PRO-1189 if the user clicks refresh, return to the same draft/published and edit/preview mode

### PR DESCRIPTION
**Pull request by boutell**

### CODE REVIEW

File: modules/@apostrophecms/admin-bar/ui/apos/components/TheAposContextBar.vue [hunk 1]
```diff
@@ -142,24 +142,17 @@ export default {
       sessionStorage.setItem('aposTabId', tabId);
     }
     window.apos.adminBar.tabId = tabId;
-
-    if (this.editMode) {
-      // Watch out for legacy situations where edit mode is active
```
Comment (boutell) [inline modules/@apostrophecms/admin-bar/ui/apos/components/TheAposContextBar.vue:147]: This is no longer a concern.

File: modules/@apostrophecms/admin-bar/ui/apos/components/TheAposContextBar.vue [hunk 2]
```diff
@@ -142,24 +142,17 @@ export default {
...
-      // but we are not in draft
-      if (!await this.lock(`${this.action}/${this.context._id}`)) {
-        this.lockNotAvailable();
-        return;
+    const lastBaseContext = JSON.parse(sessionStorage.getItem('aposLastBaseContext') || '{}');
+    if (lastBaseContext.aposDocId === this.context.aposDocId) {
+      if (lastBaseContext.draftMode !== this.draftMode) {
+        await this.setContext({ mode: lastBaseContext.draftMode });
       }
-      if (this.draftMode !== 'draft') {
```
Comment (boutell) [inline modules/@apostrophecms/admin-bar/ui/apos/components/TheAposContextBar.vue:153]: More of the dead code for a no longer extant edge case that predates the change away from storing the editing state in the session.

File: modules/@apostrophecms/admin-bar/ui/apos/components/TheAposContextBar.vue [hunk 3]
```diff
@@ -142,24 +142,17 @@ export default {
...
-
-    if (this.editMode) {
-      // Watch out for legacy situations where edit mode is active
-      // but we are not in draft
-      if (!await this.lock(`${this.action}/${this.context._id}`)) {
-        this.lockNotAvailable();
-        return;
+    const lastBaseContext = JSON.parse(sessionStorage.getItem('aposLastBaseContext') || '{}');
+    if (lastBaseContext.aposDocId === this.context.aposDocId) {
+      if (lastBaseContext.draftMode !== this.draftMode) {
```
Comment (abea) [inline modules/@apostrophecms/admin-bar/ui/apos/components/TheAposContextBar.vue:147]: Any sense in checking if `lastBaseContext.draftMode` exists? And `lastBaseContext.editMode` below?

Comment (boutell) [reply modules/@apostrophecms/admin-bar/ui/apos/components/TheAposContextBar.vue:147]: If lastBaseContext exists it'll have these properties. We don't always sanity-check every object we get from aposDocs (we trust our inserts and updates to insert well formed stuff) so I think this is OK here as well. Worst case scenario would be an unnecessary request on a one-time basis.

**Review by abea — DISMISSED** [review]

**Pull request merged**